### DISCUSSION TITLE
Create/Keep outputing to file if edited while scanning

### DISCUSSION THREAD

**Discussion by StringManolo** [projectdiscovery/nuclei#2080] [Ideas]
When editing/rm a file that's beeing used by Nuclei to direct output to, using -o flag, Nuclei will keep scanning but stop writting to the file.  
  
A --reattach-output-file (or whatever name fits) flag will be very handy to check if the output has been writed correctly in the file, and if not, reatach or recreate and write on it.
  
This feature will be very handy.
- You can manually add comentaries to the output file while scanning, like links to cve documentation  
- Manually remove entries after copy to another file, if the ouput file is getting way to big, so you can keep managing it without crashes on low memory devices  
- Make 100% sure the output is being written so you don't miss anything if another tools edit the file

**Comment by forgedhallpass** [comment]
Hello @StringManolo,

You should not alter files that are being modified by another process, because it can cause unpredictable results, synchronization issues and even potential deadlocks! The only considerable part of this feature request should be related to splitting the output into multiple equal chunks. That way you could do your processing on log files that nuclei finished writing.

p.s. Would you mind detailing your use-case, because I don't really understand what exactly you'd like to achieve. If you really want real-time processing of the log files, you could open them as read-only, do your processing and write your modifications into a separate file.

**Comment by StringManolo** [comment]
Hello, i had a problem opening the file using vim when logged into a vps using ssh.

The file was around 15-30 Megabytes of text. When you scan like 20.000 targets using -s medium, you don't know if you're going to get 10 results or 40.000

I wanted to remove the 1000 first entries of the output file  so when finish manually processing them and i get back to the output file, i don't have a bunch of old reports that i already copied.

Eventually the file got to big, that i was unable to cat | head or manage in any way (android device)

So, the scan keeps going, and i was unable to keep saving to the output file. So i has to rescan and lost 36 hours.

Since is a good program, i was thinking that would be capable of keep writing on the file, or if i deleted it, recreate it by default.

Since this behaviour will not be the best in performance, a new flag will be handy.

**Comment by StringManolo** [comment]
Closed by missclick

**Comment by forgedhallpass** [comment]
You have multiple options:
* you can control the amount of nuclei output (see the `OUTPUT` section in `nuclei -h`)
* you can split large files into smaller chunks using the `split` command
* don't use programs that load the whole file into memory! Instead you can use `grep` (without `cat`), `tail`, `head`, `less` etc. 
* or as a last resort, you can always download the log file to a more powerful machine

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
